urllib3/poolmanager.py

```diff
@@ -65,6 +65,10 @@ def __init__(self, num_pools=10, headers=None, **connection_pool_kw):
         self.pools = RecentlyUsedContainer(num_pools,
                                            dispose_func=lambda p: p.close())
 
+        # Locally set the pool classes so other PoolManagers can override them.
+        if not hasattr(self, 'pool_classes_by_scheme'):
```

Comment:
Isn't it just a matter of whether they override before they call super().__init__()?

If it is necessary, I can think of a dozen other places we might need to add it. :P